src/main/java/com/sfjs/security/ApiKeyFilter.java

```diff
@@ -0,0 +1,34 @@
+package com.sfjs.security;
+
+import java.io.IOException;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+@Component
+public class ApiKeyFilter extends OncePerRequestFilter {
+
+  private static final String AUTH_TOKEN_HEADER_NAME = "X-API-KEY";
```

Comment:
I would use the `Authorization` header for the frontend's access token.
I think `X-API-KEY` is mostly used for external API Keys that customers can get access to consume our API.

Or is this something else?

Reply:
This is generic API access protection.  It protects all endpoints, including those that do not require account ownership.
- login
- database utility
- all REST API's
and allows us to use GQL clients like Altair to view the backend directly.

The frontend includes this header for all API access, regardless of whether the account holder is logged in or not